docs/troubleshooting.md

```diff
@@ -0,0 +1,212 @@
+# Collector Troubleshooting
+
+This document describes common error conditions that Collector may encounter, and
+provides steps that can be taken to diagnose the cause, fix potential issues and
+return Collector to a working state.
+
+If the troubleshooting steps below do not solve the issue, please contact ACS
+support.
+
+## First Steps
+
+### Logs
+
+The first thing to look at are the logs from any failing Collectors. The troubleshooting
+guides below will offer examples of log messages you can expect to see for a number of
+error conditions, so having access to your own logs is invaluable.
+
+Depending on the environment, and accesses, you can get these in a number of ways.
+
+#### Kubectl
+
+Kubernetes is the easiest way to retrieve the logs, provided you have access to
+do so. 
+
+```sh
+$ kubectl get pods -n stackrox | grep collector
+collector-vclg5                      1/2     CrashLoopBackOff   2 (25s ago)   2m41s
+
+$ kubectl logs -n stackrox collector-vclg5 collector
+```
+
+#### OC
+
+Since the `oc` tool maintains compatibility with kubectl, the required commands are
+much the same.
+
+```sh
+$ oc get pods -n stackrox | grep collector
+collector-vclg5                      1/2     CrashLoopBackOff   2 (25s ago)   2m41s
+
+$ oc logs -n stackrox collector-vclg5 collector
+```
+
+#### Diagnostic Bundle
+
+An alternative method is to download a diagnostic bundle from the ACS UI, by
+following the instructions [here](https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_security_for_kubernetes/3.72/html-single/configuring/index#generate-diagnostic-bundle)
+
+Within the downloaded bundle, the logs for all Collector pods will be available
+for inspection.
+
+### Pod Status
+
+Another way of getting a quick look at the reason for a crashing Collector is to
+look at the last state of the Pod. Any failure messages are written to the last
+status, and can be viewed using `kubectl` or `oc`:
+
+```sh
+# substitute your collector pod into this command
+$ kubectl describe pod -n stackrox collector-vclg5
+[...]
+    Last State:     Terminated
+      Reason:       Error
+      Message:      No suitable kernel object downloaded
+      Exit Code:    1
+      Started:      Fri, 21 Oct 2022 11:50:56 +0100
+      Finished:     Fri, 21 Oct 2022 11:51:25 +0100
+[...]
+```
+
+We can instantly see that the Collector has failed to download a kernel driver.
+
+## Collector Start Up
+
+The vast majority of errors occur during Collector startup, where Collector will
+configure itself, find or download an appropriate kernel driver for the system,
+and then start Collecting events. The following diagram describes the main parts
+of this start-up process:
+
+<p align="center">
+  <img src="./imgs/collector-startup.png" />
+</p>
```

Comment:
This diagram shows that collector startup could fail at "Analyse host". Perhaps this document could show errors at this point. Also the American spelling, analyze, should probably be used.